File: app/src/androidTest/java/br/com/jonathanzanella/myexpenses/account/AccountRepositoryTest.java
package br.com.jonathanzanella.myexpenses.account;

import android.support.test.filters.SmallTest;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import javax.inject.Inject;

import br.com.jonathanzanella.TestApp;
import br.com.jonathanzanella.myexpenses.App;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.not;

@RunWith(AndroidJUnit4.class)
@SmallTest
public class AccountRepositoryTest {
	@Inject
	AccountDataSource dataSource;

	@Before
	public void setUp() throws Exception {
		TestApp.Companion.getTestComponent().inject(this);
		App.Companion.resetDatabase();
	}

	@Test
	public void can_save_account() throws Exception {
		Account account = new Account();
		account.setName("test");
		dataSource.save(account);

		assertThat(account.getId(), is(not(0L)));
		assertThat(account.getUuid(), is(not("")));
	}

	@Test
	public void can_load_saved_account() throws Exception {
		Account account = new Account();
		account.setName("test");
		dataSource.save(account);

		Account loadAccount = dataSource.find(account.getUuid()).blockingFirst();
		assertThat(loadAccount.getUuid(), is(account.getUuid()));
	}
}

File: app/src/androidTest/java/br/com/jonathanzanella/myexpenses/account/AccountRepositoryTest.kt
package br.com.jonathanzanella.myexpenses.account

import android.support.test.filters.SmallTest
import android.support.test.runner.AndroidJUnit4
import br.com.jonathanzanella.TestApp
import br.com.jonathanzanella.myexpenses.App
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.not
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import javax.inject.Inject

@RunWith(AndroidJUnit4::class)
@SmallTest
class AccountRepositoryTest {
    @Inject
    lateinit var dataSource: AccountDataSource

    @Before
    @Throws(Exception::class)
    fun setUp() {
        TestApp.getTestComponent().inject(this)
        App.resetDatabase()
    }

    @Test
    @Throws(Exception::class)
    fun can_save_account() {
        val account = Account()
        account.name = "test"
        dataSource.save(account).subscribe {
            assertThat(account.id, `is`(not(0L)))
            assertThat<String>(account.uuid, `is`(not("")))
        }
    }

    @Test
    @Throws(Exception::class)
    fun can_load_saved_account() {
        val account = Account()
        account.name = "test"
        dataSource.save(account).subscribe {
            dataSource.find(account.uuid!!).subscribe {
                assertThat<String>(it.uuid, `is`<String>(account.uuid))
            }
        }
    }
}